lua/nvim-tree/trash.lua

```diff
@@ -0,0 +1,73 @@
+local M = {}
+
+local lib = require'nvim-tree.lib'
+local utils = require'nvim-tree.utils'
+local events = require'nvim-tree.events'
+local fs = require'nvim-tree.fs'
+local api = vim.api
+
+local function clear_buffer(absolute_path)
+  local bufs = vim.fn.getbufinfo({bufloaded = 1, buflisted = 1})
+  for _, buf in pairs(bufs) do
+    if buf.name == absolute_path then
+      if buf.hidden == 0 and #bufs > 1 then
+        local winnr = api.nvim_get_current_win()
+        api.nvim_set_current_win(buf.windows[1])
+        vim.cmd(':bn')
+        api.nvim_set_current_win(winnr)
+      end
+      vim.api.nvim_buf_delete(buf.bufnr, {})
+      return
+    end
+  end
+end
+
+function M.trash_node(node, cfg)
+  if node.name == '..' then return end
+
+  -- configs
+  if cfg.is_unix then
+    if cfg.trash.cmd == nil then cfg.trash.cmd = 'trash' end
+    if cfg.trash.require_confirm  == nil then cfg.trash.require_confirm  = true end
+  else
+    print('trash is currently a UNIX only feature!')
+  end
+
+  -- trashes a path (file or folder)
+  local function trash_path(on_exit)
+    vim.fn.jobstart(cfg.trash.cmd.." "..node.absolute_path, {
+      detach = true,
+      on_exit = on_exit(),
```

Comment:
this will be called when trash_path is called, not when the job finishes. just remove the parens :)

Reply:
ah, that makes sense, should be good now.